specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-07-01-preview/virtualHardDisks.json

```diff
@@ -0,0 +1,428 @@
+        "x-ms-examples": {
+          "ListVirtualHardDiskByResourceGroup": {
+            "$ref": "./examples/ListVirtualHardDiskByResourceGroup.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Lists all virtualharddisks under the resource group.",
+            "schema": {
+              "$ref": "#/definitions/virtualharddisksListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.AzureStackHCI/virtualharddisks": {
+      "get": {
+        "operationId": "virtualharddisks_ListBySubscription",
+        "x-ms-examples": {
+          "ListVirtualHardDiskByResourceGroup": {
+            "$ref": "./examples/ListVirtualHardDiskBySubscription.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Lists all virtualharddisks under the subscription.",
+            "schema": {
+              "$ref": "#/definitions/virtualharddisksListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "virtualharddisksProperties": {
+      "type": "object",
+      "properties": {
+        "blockSizeBytes": {
+          "type": "integer",
+          "format": "int32"
+        },
+        "diskSizeBytes": {
+          "type": "integer",
+          "format": "int64",
+          "description": "diskSizeBytes - size of the disk in GB"
+        },
+        "dynamic": {
```

Comment:
>dynamic [](http://example.com/codeflow?start=9&length=7)

Prefer string enum vs. boolean. Ref: [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean#common-issues-found-in-review). Could use sizingType: { Static, Dynamic }

Reply:
The API was published to our customers with the "boolean" type (this API was copied from RPSaaSdev, where it was reviewed before). In order to avoid introducing any breaking change, we'd like to keep "boolean" in this version of the API and fix this in a future revision. Does this plan sound good?

Reply:
Added this workitem to this pending task: https://dev.azure.com/msazure/msk8s/_boards/board/t/ARC-HCI/Stories/?workitem=12679581